### DISCUSSION TITLE
How to build a standalone executable with postcss-nested bundled?

### DISCUSSION THREAD

**Discussion by hudri** [tailwindlabs/tailwindcss#10709] [Help] [answered]
I want to use TailwindCSS in an environment without NodeJS, so I use the single-file binary executable. I also want to use `postcss-import` and `postcss-nested`, but since there is no NodeJS on this server, I cannot use a custom `postcss.config.js`.

I therefore created my own fork of TailwindCSS, and [modified the default postcss.config.js stub to include postcss-import](https://github.com/webtourismus/tailwindcss-standalone-bundler/blob/master/stubs/defaultPostCssConfig.stub.js). Then using your Github "Prepare release" action in my own fork, I created [my own single-file executables with postcss-import bundled](https://github.com/webtourismus/tailwindcss-standalone-bundler/releases/tag/untagged-e280f940eb9dd5ff5017). This worked as expected, imported CSS files are now inlined when using my custom executable.

I then tried the same trick again and also added `tailwindcss/nesting` to the postcss-config-stub, and sadly this did not work. I can see my nested CSS rules in the output, but they are not transformed to native CSS selectors, but rendered as written in the input CSS.

Can you give me some hints how to include postcss-nested and tailwindcss/nesting into my own executable?

**Comment by hudri** [comment, marked as answer]
I've finally managed to create my own executable. If anyone is interested, here are some links how I did it:

https://github.com/tailwindlabs/tailwindcss/compare/master...webtourismus:tailwindcss:patched

With the repo above as helper, I created a composer package so I can use TailwindCSS on my PHP environment without NodeJS:

https://packagist.org/packages/webtourismus/tailwindcss-cli

**Comment by drzraf** [comment]
What's the current status. Could the self-contained binary include PostCSS out of the box nowadays?
